src/settings/window_geometry.rs

```diff
@@ -1,16 +1,73 @@
-#[derive(Debug, Clone)]
+use crate::renderer::Renderer;
+use crate::settings::SETTINGS;
+use crate::window::WindowSettings;
+use glutin::dpi::PhysicalSize;
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, Copy)]
 pub struct WindowGeometry {
     pub width: u64,
     pub height: u64,
 }
 
-pub const DEFAULT_WINDOW_GEOMETRY: WindowGeometry = WindowGeometry {
+pub fn try_to_load_last_window_size() -> Result<WindowGeometry, String> {
+    let mut settings_path = dirs::home_dir().unwrap();
+    settings_path.push(".local/share/nvim/neovide-settings.txt");
+    let serialized_size = std::fs::read_to_string(&settings_path).map_err(|e| e.to_string())?;
+
+    let de: PhysicalSize<u32> =
+        serde_json::from_str(&serialized_size).map_err(|e| e.to_string())?;
+    log::debug!("Loaded Window Size: {:?}", de);
+    Ok(WindowGeometry {
+        width: de.width as u64,
+        height: de.height as u64,
+    })
+}
+#[cfg(unix)]
+fn neovim_std_datapath() -> PathBuf {
+    let mut settings_path = dirs::home_dir().unwrap();
+    settings_path.push(".local/share/nvim/neovide-settings.txt");
+    settings_path
+}
+#[cfg(windows)]
+fn neovim_std_datapath() -> PathBuf {
+    let mut settings_path = dirs::home_dir().unwrap();
+    settings_path.push("AppData/Local/nvim-data/neovide-settings.txt");
+    settings_path
+}
+
+const PRIVATE_WINDOW_GEOMETRY: WindowGeometry = WindowGeometry {
     width: 100,
     height: 50,
 };
 
+lazy_static! {
+    pub static ref DEFAULT_WINDOW_GEOMETRY: WindowGeometry = try_to_load_last_window_size()
+        .or::<String>(Ok(PRIVATE_WINDOW_GEOMETRY))
+        .unwrap();
+}
+
+pub fn maybe_save_window_size(window_size: PhysicalSize<u32>, renderer: &Renderer) {
+    let saved_window_size = if SETTINGS.get::<WindowSettings>().remember_dimension {
+        PhysicalSize::<u32> {
+            width: (window_size.width as f32 / renderer.font_width as f32) as u32,
+            height: (window_size.height as f32 / renderer.font_height as f32) as u32,
+        }
+    } else {
+        PhysicalSize::<u32> {
+            width: PRIVATE_WINDOW_GEOMETRY.width as u32,
```

Comment:
I don't think we should save default values.

If remember_window_size is false, this function should do nothing.

Reply:
The problem with that is that after the user changes the "remember_window_size" variable back to false, neovide will still use the previously saved window setting instead of the default.

Reply:
Oh, i see.

And we probably can't check that variable here, becouse remember_window_size is not initialized at the moment of reading cmdline settings?

Maybe it's better to move it to env variable for the start, and then if it works fine, remove variable and just do that by default. Other apps just do it that way.

Let's wait what @Kethku has to say about this.

Reply:
The issue with that approach is if the user changes `remember_window_size` back to false, they will still get the previously saved window size instead of the default upon restart.

Reply:
I think I'm fine with this behavior until we restructure the app to wait for the neovim connection before popping the window.